pennylane/_version.py

```diff
@@ -16,4 +16,4 @@
 Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.39.0-dev45"
+__version__ = "0.39.0"
```

Comment:
```suggestion
__version__ = "0.39.0rc0"
```
https://peps.python.org/pep-0440/#pre-releases

To avoid the nightly pushes going to test PyPI with the release version.

Reply:
This PR is only to set up the RC branch. Once it's merged and the RC branch is created, there will be an immediate 2nd PR to bumpy the version to `0.40.0-dev0`. Is it still necessary to add the `.rc0` in that case?